### PR TITLE
Whitelist Support Interface

### PR DESCRIPTION
**Pull request by wilmerrootstock**
## Description
Create an interface called WhitelistSupport, and put all the method signatures Whitelist-related on it. Take these methods from BridgeSupport. The plan is to remove all of them from BridgeSupport into their own WhitelistSupport class.

## Motivation and Context
This relates to the Bridge Refactor to decouple some objects tied closely to the Bridge.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Requires Activation Code (Hard Fork)

* **Other information**:

### CODE REVIEW

File: rskj-core/src/main/java/co/rsk/peg/whitelist/WhitelistSupport.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+import java.math.BigInteger;
+import org.ethereum.core.Transaction;
+
+/**
+ * Interface for the whitelist support. This interface is used to add, remove and get whitelist entries.
+ */
+public interface WhitelistSupport {
+
+    Integer getLockWhitelistSize();
```
Comment (apancorb) [inline rskj-core/src/main/java/co/rsk/peg/whitelist/WhitelistSupport.java:11]: Can the methods that return a boxed primitive type be changed to just a primitive instead? This way we avoid doing null checks.

File: rskj-core/src/main/java/co/rsk/peg/whitelist/WhitelistSupport.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+    LockWhitelistEntry getLockWhitelistEntryByIndex(int index);
+
+    LockWhitelistEntry getLockWhitelistEntryByAddress(String addressBase58);
```
Comment (apancorb) [inline rskj-core/src/main/java/co/rsk/peg/whitelist/WhitelistSupport.java:15]: Will these get methods return a `null` reference in the case of not being able to retrieve the `LockWhitelistEntry`? What about returning an `Optional<LockWhitelistEntry>` and if the get fails return an `Optional.empty()`?

Comment (marcos-iov) [reply rskj-core/src/main/java/co/rsk/peg/whitelist/WhitelistSupport.java:15]: That's a good point, I'll create a separate task to do that change

File: rskj-core/src/main/java/co/rsk/peg/whitelist/WhitelistSupport.java (hunk 1 above)
Comment (marcos-iov) [reply rskj-core/src/main/java/co/rsk/peg/whitelist/WhitelistSupport.java:11]: Agree

**Review by marcos-iov — APPROVED** [review]

**Pull request merged**